sdk/core/core/src/az_span.c

```diff
@@ -557,6 +557,9 @@ AZ_NODISCARD az_result az_span_i64toa(az_span destination, int64_t source, az_sp
     return _az_span_builder_append_uint64(out_span, (uint64_t)-source);
   }
 
+  // make out_span point to destination before trying to write on it (might be an empty az_span or
+  // pointing else where)
+  *out_span = destination;
```

Comment:
> some of the issues coming from last code refactor. Found while running storage sample.

Just to clarify, this API had the bug before the redesign, correct? None of the logic was changed.

https://github.com/Azure/azure-sdk-for-c/blob/fa4d3a1e9402ed2fd79b0e13916be951b2fd1735/sdk/core/core/src/az_span.c#L480-L491